packages/core/data-transfer/lib/engine/index.ts

```diff
@@ -51,10 +54,36 @@ class TransferEngine implements ITransferEngine {
     }
 
     throw new Error(
-      `Strapi versions doesn't match (${strategy} check): ${sourceVersion} does not match with ${destinationVersion} `
+      `Strapi versions doesn't match (${strategy} check): ${sourceVersion} does not match with ${destinationVersion}`
     );
   }
 
+  #assertSchemasMatching(sourceSchemas: any, destinationSchemas: any) {
+    const chosenStrategy = this.options.schemasMatching || 'strict';
```

Comment:
I don't think that this option can be undefined (in the defined typing).
If you have an undefined value for some reason, it might be because we need to set the value in the export/import commands instead.

Reply:
I disagree here, it's ok to have the extra check, especially since we may have to remove typescript. And even with typescript, we have a lot of js in the codebase (including the import/export commands from the CLI) that could potentially call it incorrectly.

I could see turning it into an assertion rather than a default, but I don't have a strong opinion either way.